prow/plugins/config.go

```diff
@@ -1049,6 +1050,39 @@ type BugzillaRepoOptions struct {
 	Branches map[string]BugzillaBranchOptions `json:"branches"`
 }
 
+// BugzillaBugState describes bug states in the Bugzilla plugin config, used
+// for example to specify states that bugs are supposed to be in or to which
+// they should be made after some action.
+type BugzillaBugState struct {
+	Status     string `json:"status,omitempty"`
+	Resolution string `json:"resolution,omitempty"`
+}
+
+// AsBugUpdate converts a state into a BugUpdate struct used by the API
+// to update bugs
+func (state *BugzillaBugState) AsBugUpdate() *bugzilla.BugUpdate {
+	if state == nil {
+		return nil
+	}
+
+	return &bugzilla.BugUpdate{Status: state.Status, Resolution: state.Resolution}
+}
+
+// Matches returns whether a given bug matches the state
+func (state *BugzillaBugState) Matches(bug *bugzilla.Bug) bool {
```

Comment:
Why do we have `bugMatchesState` if we have this?

Reply:
It's `bugMatchesStateS` which is true is `state.Matches(bug)` for at least one state in a slice. Makes callsites better readable.

Reply:
:man_facepalming:  poor reading on my end